### PR TITLE
Setup the `deployWorkspaceProject` test flow and hook up initial tests for monorepo

### LINKED ISSUE #698: Add a general framework for testing projects (`deployWorkspaceProject`)

### LINKED ISSUE #704: Add basic test coverage for monorepos

### PR DESCRIPTION
**Pull request by MicroFish91**
Overall todos outlined here: https://github.com/microsoft/vscode-azurecontainerapps/issues/697

Closes #704 
Closes #698

### High-Level Overview

1. Define a list of test scenarios (which is basically just a logical grouping for a sequence of test cases):
image

2. Each scenario can have multiple test cases:
image

3. Loop through each scenario's test cases in sequential order.  After each case, verify expected output matches what was defined in the test case.  Core logic implementation for automatically running through each test scenario/test case found under `deployWorkspaceProject.test.ts`

### CODE REVIEW

File: test/nightly/deployWorkspaceProject/testScenarios.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type DeployWorkspaceProjectTestCase } from "./testCases/DeployWorkspaceProjectTestCase";
+import { generateMonoRepoBasicTestCases } from "./testCases/monoRepoBasicTestCases";
+
+interface TestScenario {
```
Comment (alexweininger) [inline test/nightly/deployWorkspaceProject/testScenarios.ts:9]: Seems like these test scenarios are not as generic as the name implies. Should it be renamed to `DeployWorkspaceProjectTestScenario`?

I guess what's confusing is that within the generic test scenario, the only property for test cases is specific to deploying a workspace project.

Comment (MicroFish91) [reply test/nightly/deployWorkspaceProject/testScenarios.ts:9]: Yeah good catch, I should have used that as the name here.

File: test/test.code-workspace [hunk 2]
```diff
@@ -7,6 +7,10 @@
         {
             "name": "monorepo-basic",
             "path": "./testProjects/monorepo-basic"
-        }
+        },
+        // {
+        //     "name": "albumapi-javascript",
+        //     "path": "./testProjects/albumapi-javascript"
+        // },
```
Comment (alexweininger) [inline test/test.code-workspace:14]: I'm assuming this will be uncommented in a later PR

File: test/nightly/global.nightly.test.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    // Account for the fact that it can take an extremely long time to delete managed environment resources
+    this.timeout(30 * 60 * 1000);
+    await deleteResourceGroups();
+});
+
+async function deleteResourceGroups(): Promise<void> {
+    const context: TestActionContext = await createTestActionContext();
+    const rgClient: ResourceManagementClient = createAzureClient([context, subscriptionContext], ResourceManagementClient);
+
+    await Promise.all(Array.from(resourceGroupsToDelete).map(async resourceGroup => {
```
Comment (alexweininger) [inline test/nightly/global.nightly.test.ts:45]: I think `Promise.allSettled` would be best here. Unlike `Promise.all`, it doesn't reject as soon as one promise rejects, it waits until all promises have either resolved or rejected.

**Review by alexweininger — APPROVED** [review]

**Pull request merged**